src/basic/process-util.c

```diff
@@ -2100,6 +2100,15 @@ int posix_spawn_wrapper(
         _cleanup_close_ int pidfd = -EBADF;
 
         r = pidfd_spawn(&pidfd, path, NULL, &attr, argv, envp);
+        if (r == E2BIG) {
+                /* Some kernels (e.g., 5.4) support clone3 but they do not support CLONE_INTO_CGROUP.
+                 * Retry pidfd_spawn() after removing the flag. */
+                flags &= ~POSIX_SPAWN_SETCGROUP;
+                r = posix_spawnattr_setflags(&attr, flags);
+                if (r != 0)
+                        return -r;
+                r = pidfd_spawn(&pidfd, path, NULL, &attr, argv, envp);
+        }
```

Comment:
and please, as suggested earlier, cache the result, i.e. add "static bool setcgroup_supported = true;" somewhere as a local static variable here, and then set it to false if we realize it doesn't actually work. Then, always check this field before we try to do anything.

Reply:
OK, did so in a second patch. I only update the static variable if we were successful because otherwise, there might be something else going wrong.